src/main/java/de/tum/in/www1/artemis/config/websocket/WebsocketConfiguration.java

```diff
@@ -268,6 +270,32 @@ public Message<?> preSend(@NotNull Message<?> message, @NotNull MessageChannel c
          * @return flag whether subscription is allowed
          */
         private boolean allowSubscription(Principal principal, String destination) {
+            User user = userRepository.getUserWithGroupsAndAuthorities(principal.getName());
+            if (destination.equals("/topic/admin/queued-jobs") || destination.equals("/topic/admin/running-jobs")) {
```

Comment:
just a thought, but what happens if destination is `null`? can that be? if so, maybe switch to `"/topic/admin/queued-jobs".equals(destination)`